### PR TITLE
Feat/add validation summary

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé

This PR adds a WTForms-based `validation_summary` macro to allow display of an accessible validation summary component that:
- focuses the browser on the summary when its presented
- enumerates the errors on the form
- provides links to each error 

## Details
This PR adds a new `validation_summary` component which can be used on any form to display a linked summary of errors on that form.  It also updates the `form_wrapper` macro to add 2 new parameters (`show_validation_summary` and `form`) that, when provided, automatically invoke the `validation_summary` component.

## Related cards
- https://github.com/cds-snc/notification-planning/issues/1470

## Before
image

## After
image

# Test instructions | Instructions pour tester la modification

- Go to the `/register` page
- Attempt to register without satisfying all of the form fields
- Ensure missing/incorrect fields are part of the summary
- Ensure you can click on each link in the summary and are brought to the corresponding field

### CODE REVIEW

**Comment by amazingphilippe** [comment]
I noticed a bug while styling the summary: When an error message contains a list, it adds them in the summary and we lose the context of the problem :(
image

**Comment by andrewleith** [comment]
> I noticed a bug while styling the summary: When an error message contains a list, it adds them in the summary and we lose the context of the problem :( image

I'm sure I can fix that - how do you reproduce it with multiple errors like that?

**Comment by andrewleith** [comment]
Oh nevermind I see - we stuffed a bunch of `<li>`'s into the error message. We actually aren't creating valid markup here in general :/

**Review by amazingphilippe — CHANGES_REQUESTED** [review]
Just found one more case where we might want to handle html a bit more carefully: 

image

I wonder if we can use a filter to sanitize the html so that we get a plain text value to show. Or we edit this error message in particular so that it doesn't have a link in it. Contacting us should be straightforward enough to not provide a link there.

**Comment by andrewleith** [comment]
> I wonder if we can use a filter to sanitize the html so that we get a plain text value to show. Or we edit this error message in particular so that it doesn't have a link in it. Contacting us should be straightforward enough to not provide a link there.

Hmm yeah I feel like we are not using the WTForms validation errors correctly by including HTML in them.  I would lean towards stopping to put complex HTML in the error messages.

File: app/main/forms.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def password(label=_l("Password")):
             Length(8, 255, message=_l("Must be at least 8 characters")),
             Blocklist(
                 message=_l(
-                    "A password that is hard to guess contains:<li>Uppercase and lowercase letters.</li><li>Numbers and special characters.</li><li>Words separated by a space.</li>"
+                    "A password that is hard to guess contains:<ul class='list list-bullet'><li>Uppercase and lowercase letters.</li><li>Numbers and special characters.</li><li>Words separated by a space.</li></ul>"
```
[the remark this answers is not in the document]
Comment (andrewleith) [reply app/main/forms.py:170]: Yeah that would certainly work better!  I can make that change, too!

File: app/main/forms.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ def password(label=_l("Password")):
             Length(8, 255, message=_l("Must be at least 8 characters")),
             Blocklist(
                 message=_l(
-                    "A password that is hard to guess contains:<li>Uppercase and lowercase letters.</li><li>Numbers and special characters.</li><li>Words separated by a space.</li>"
+                    "A password that is hard to guess contains: <ul class='list list-bullet'><li>Uppercase and lowercase letters.</li><li>Numbers and special characters.</li><li>Words separated by a space.</li></ul>"
```
Comment (amazingphilippe) [inline app/main/forms.py:170]: Don't forget to update this one too!

**Review by amazingphilippe — APPROVED** [review]
Looks good to me! Really awesome work. Can't wait to try it on other forms! ⭐

**Pull request merged**